File: RadarHub.Back/RadarHub.API/Controllers/FonteOrcamentariaController.cs
using Microsoft.AspNetCore.Mvc;
using RadarHub.Dominio.Entidades;
using RadarHub.Dominio.Servicos;
using RSK.API.Controllers;
using RSK.Dominio.Interfaces;
using RSK.Dominio.Notificacoes.Interfaces;

namespace RadarHub.API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class FonteOrcamentariaController : ApiConsultaControllerBase<FonteOrcamentaria>
    {
        private readonly FonteOrcamentariaServico _tipoServico;
        private readonly INotificador _mensagens;

        public FonteOrcamentariaController(IServicoConsultaBase<FonteOrcamentaria> servicoConsulta, FonteOrcamentariaServico tipoServico, INotificador mensagens) : base(servicoConsulta)
        {
            _tipoServico = tipoServico;
            _mensagens = mensagens;
        }


        [HttpPost("Importar")]
        public async Task<IActionResult> Importar()
        {
            var sucesso = await _tipoServico.ImportarAsync();
            var resultado = _mensagens.ObterMensagens();

            if (!sucesso)
            {
                return BadRequest(resultado);
            }

            return Ok(resultado);
        }
    }
}

File: RadarHub.Web/src/servicos/FonteOrcamentariaController.ts
import { FonteOrcamentaria } from "../DTO/FonteOrcamentaria";
import { ConsultaServicoBase } from "./ConsultaServicoBase";
import http from "./http-comon"

class FonteOrcamentariaController extends ConsultaServicoBase<FonteOrcamentaria> {
    constructor() {
        super("FonteOrcamentaria")
    }

    public async importar() {
        return http.post(`/fonteorcamentaria/importar`)
    }
}

const FonteOrcamentariaServico = new FonteOrcamentariaController()

export default FonteOrcamentariaServico;